sql-db/sql-app/src/test/resources/mysql_import.sql

```diff
@@ -0,0 +1,8 @@
+INSERT INTO book (id, title, author) VALUES (1, 'Foundation', 'Isaac Asimov');
```

Comment:
No such thing in MySQL. We can use select query to Book_SEQ table if deem it better. Personally I don't feel the appeal.

Reply:
I have limited knowledge here, but the way I understand it, sequence there works as transaction - you select value for update from sequence table and than you update the value to the next value. I don't know why should we do it for every insert as there is no interference with other DB users (e.g. other app).

Reply:
Ok, it's possible. I'll do it.

Reply:
Well, with MySQL is simple when you create a table you just have to define your ID as `AUTO_INCREMENT`, that is all! :)

Now the question is, how can I do that, with Hibernate?

Reply:
Table strategy, but we can't have single model for all the databases that uses sequence strategy and only change it for MySQL. Also I think table strategy is not recommended for other databases.

Reply:
Ok I think that I found the way to do it in hibernate:

On your entity define you ID by this way:

```
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
```

And don´t forget to define the dialect into the application.properties`
mysql.properties
```
quarkus.hibernate-orm.dialect=org.hibernate.dialect.MySQLDialect
```

postgres.properties
```
quarkus.hibernate-orm.dialect=org.hibernate.dialect.PostgresPlusDialect
```

Finally remove the ID from your inserts (because is going to be auto-generated)

Mysql import SQL:
```
INSERT INTO book (title, author) VALUES ('Foundation', 'Isaac Asimov');
INSERT INTO book (title, author) VALUES ('2001: A Space Odyssey', 'Arthur C. Clarke');
INSERT INTO book (title, author) VALUES ('Stranger in a Strange Land', 'Robert A. Heinlein');
INSERT INTO book (title, author) VALUES ('Ender''s Game', 'Orson Scott Card');
INSERT INTO book (title, author) VALUES ('Hyperion', 'Dan Simmons');
INSERT INTO book (title, author) VALUES ('Anathem', 'Neal Stephenson');
INSERT INTO book (title, author) VALUES ('Perdido Street Station', 'China Miéville');
```

I didn´t test all the datasources but at least is working for postgres and Mysql

Reply:
I know identity, but I am missing your point. Rostislav wanted this to be closer to the real life application. Except for MySQL, sequence generator is recommended. With identity generator,  Hibernates disables the JDBC batch support for entities.

Reply:
Ok, let's keep it as it is. Oracle was the only datastore that doesn´t support "auto-increment" feature

Reply:
good to know. well, I happen to remember we used identity till we needed batch inserts, so that's why I don't like them :-)